src/keyboard.js

```diff
@@ -0,0 +1,288 @@
+import { Key } from './key';
+import { LINE_BREAK, KEY_LAYOUTS, SPECIALKEYS } from './keyLayouts';
+import { SpecialActions } from './specialActions';
+
+export class Keyboard {
+  constructor(textArea, keyboardPainter) {
+    this.keyboardPainter = keyboardPainter;
+    this.textArea = textArea;
+    this.isShiftActive = false;
+    this.properties = {
+      value: "",
+      capsLock: false,
+      language: null,
+      source: [],
+      pressed: new Set()
```

Comment:
Instead of using this Set you can use native functionality.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent
As you can notice KeyboardEvent has `ctrlKey`, `altKey`, `shiftKey` and `repeat` properties which helps to handle special cases.

Reply:
thanks! Read it later :)